server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/StarTreesBuilder.java

```diff
@@ -35,61 +36,52 @@ public class StarTreesBuilder implements Closeable {
     private static final Logger logger = LogManager.getLogger(StarTreesBuilder.class);
 
     private final List<StarTreeField> starTreeFields;
-    private final StarTreeFieldConfiguration.StarTreeBuildMode buildMode;
-    private final DocValuesConsumer docValuesConsumer;
     private final SegmentWriteState state;
     private final Map<String, DocValuesProducer> fieldProducerMap;
     private final MapperService mapperService;
 
     public StarTreesBuilder(
-        List<StarTreeField> starTreeFields,
-        StarTreeFieldConfiguration.StarTreeBuildMode buildMode,
         Map<String, DocValuesProducer> fieldProducerMap,
-        DocValuesConsumer docValuesConsumer,
         SegmentWriteState segmentWriteState,
         MapperService mapperService
     ) {
+        List<StarTreeField> starTreeFields = new ArrayList<>();
+        for (CompositeMappedFieldType compositeMappedFieldType : mapperService.getCompositeFieldTypes()) {
+            if (compositeMappedFieldType instanceof StarTreeMapper.StarTreeFieldType) {
+                StarTreeMapper.StarTreeFieldType starTreeFieldType = (StarTreeMapper.StarTreeFieldType) compositeMappedFieldType;
+                starTreeFields.add(new StarTreeField(starTreeFieldType.name(), starTreeFieldType.getDimensions(), starTreeFieldType.getMetrics(), starTreeFieldType.getStarTreeConfig()));
+            }
+        }
+
         this.starTreeFields = starTreeFields;
-        if (starTreeFields == null || starTreeFields.isEmpty()) {
-            throw new IllegalArgumentException("Must provide star-tree builder configs");
+        if (starTreeFields.isEmpty()) {
```

Comment:
Lets not throw exception - if present build otherwise, log it and lets return. 

If we need to throw exception ideally we should get list <starTreeFields > , composite fields might or might not contain star tree fields.

we can pick either option